src/main/java/graphql/nadel/schema/SchemaTransformation.java

```diff
@@ -0,0 +1,45 @@
+package graphql.nadel.schema;
+
+import graphql.PublicSpi;
+import graphql.schema.GraphQLSchema;
+
+/**
+ * High level representation of a a graphql schema transformation. Warning: the schema resulting from the transformation
+ * is not validated by Nadel, so may produce unpredictable results if incorrect.
+ *
+ * <p>Example usage, to delete a field:
+ * <code>
+ *     SchemaTransformation transformation = new SchemaTransformation() {
+ *         @Override
+ *         GraphQLSchema apply(GraphQLSchema originalSchema) {
+ *             return SchemaTransformer.transformSchema(originalSchema, new GraphQLTypeVisitorStub() {
+ *                  @Override
+ *                  TraversalControl visitGraphQLFieldDefinition(GraphQLFieldDefinition node, TraverserContext<GraphQLSchemaElement> context) {
+ *                      if (node.getName() == "secretField") {
+ *                          return TreeTransformerUtil.deleteNode(node);
+ *                      }
+ *
+ *                      return TraversalControl.CONTINUE;
+ *                  }
+ *             }
+ *         }
+ *     }
+ * </code>
+ *
+ * @see graphql.schema.SchemaTransformer
+ * @see graphql.schema.GraphQLTypeVisitorStub
+ */
+@PublicSpi
+public interface SchemaTransformation {
```

Comment:
Naming is hard.  Can we consider calling this `SchemaTransformationHook`

Its like the other hook we have and its a callback. We can start to name out SPI callbacks "hooks" say

This is confusingly named to the underlying SchemaTransformation in graphql-java.

Reply:
Yeah that makes sense. Will rename.

Reply:
👍  for the rename